src/webserver/http.rs

```diff
@@ -434,25 +433,89 @@ async fn serve_file(
         })
 }
 
+/// Fallback handler for when a file could not be served
+///
+/// Recursively traverses upwards in the request's path, looking for a `404.sql` to call as the
+/// fallback handler. Fails if none is found, or if there is an issue while running the `404.sql`.
+async fn serve_fallback(
+    service_request: &mut ServiceRequest,
+    original_err: actix_web::Error,
+) -> actix_web::Result<HttpResponse> {
+    let catch_all = "404.sql";
+
+    let req_path = req_path(&service_request);
+    let mut fallback_path_candidate = req_path.clone().into_owned();
+    log::debug!("Trying to find a {catch_all:?} handler for {fallback_path_candidate:?}");
+
+    let app_state: &web::Data<AppState> = service_request.app_data().expect("app_state");
+
+    // Find the indeces of each char which follows a directroy separator (`/`). Also consider the 0
+    // index, as we also have to try to check the empty path, for a root dir `404.sql`.
+    for idx in req_path
```

Comment:
Thinking about security here: could an attacker dos us by sending a request that is just thousands of slashes? Maybe we should add a (large) limit to the number of iterations?

Reply:
Whats special about this condition? The same issue arises when a passing a thousand get parameters, except that the parsing is more expensive for GET parameters, while for the fallback the file syscalls are more expensive. I get the idea, but I'm against starting to do this here. Either there is a generic way that is implemented in the actix-web to do this kind of DDoS protection, or there isn't, but I don't think it is clever to start limiting at arbitrary places by hand. Also, I'd assume that there already is a max header size, which naturally will limit the amount of iterations done here anyway.

Reply:
Yes, indeed, I was thinking about doing thousands of syscalls, or more importantly, database queries since the files can be stored in the database. I'll add the limit.

Reply:
If you add a limit, make it configurable please